### PR TITLE
HAT support for dynamic shared memory allocation for GPU

### PR DESCRIPTION
**Pull request by CaptnJackSparrow**

### CODE REVIEW

File: schema/hat.tosd [hunk 1]
```diff
@@ -101,6 +101,11 @@ version = "0.0.0.3"
         type = "array"
         optional = true
 
+        # The dynamic shared memory size in bytes to be allocated for this device function
```
Comment (lisaong) [inline schema/hat.tosd:104]: can rev the version field in line 3 (although we're not checking for version and this is optional)

**Review by lisaong — APPROVED** [review]

File: schema/hat.tosd (hunk 1 above)
Comment (CaptnJackSparrow) [reply schema/hat.tosd:104]: so it doesn't matter right?

Comment (lisaong) [reply schema/hat.tosd:104]: yeah can punt, not a hard requirement for now, as we don't check for version, and this change adds a field.

Comment (CaptnJackSparrow) [reply schema/hat.tosd:104]: skipping based on conversation...

**Pull request merged**